src/routes/docs/tutorials/angular/step-4/+page.markdoc

```diff
@@ -0,0 +1,132 @@
+---
+layout: tutorial
+title: Add authentication
+description: Add authentication to your angular application.
+step: 4
+---
+
+# User service {% #user-service %}
+
+In Angular, services are used to share data and functions between components. We'll create a UserService to manage our user's data.
+
+Create a new file `src/app/user.service.js` and add the following code to it.
+
+```js
+import { Injectable } from '@angular/core';
+import { BehaviorSubject } from 'rxjs';
+import { account } from '../appwrite';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class UserService {
+  private userSubject = new BehaviorSubject(null);
+  user$ = this.userSubject.asObservable();
+
+  async login(email, password) {
+    const loggedIn = await account.createEmailSession(email, password);
+    this.userSubject.next(loggedIn);
+  }
+
+  async logout() {
+    await account.deleteSession('current');
+    this.userSubject.next(null);
+  }
+
+  async register(email, password) {
+    await account.create(email, password);
+    await this.login(email, password);
+  }
+
+  async init() {
+    try {
+      const loggedIn = await account.get();
+      this.userSubject.next(loggedIn);
+    } catch (err) {
+      this.userSubject.next(null);
+    }
+  }
+}
+```
+
+# Basic routing  {% #basic-routing %}
+
+Angular provides a powerful router that allows you to configure navigation between components. 
+
+Update your `src/app/app-routing.module.js` to define your routes.
+
+```js
+import { NgModule } from '@angular/core';
+import { RouterModule, Routes } from '@angular/router';
```

Comment:
Make sure you show where to add the router module imports in the AppsModules like:

```ts
import { FormsModule } from '@angular/forms';
...
@NgModule({
  declarations: [
    // ...
  ],
  imports: [
    // ...
    FormsModule
  ],
  providers: [],
  bootstrap: [AppComponent]
})
export class AppModule { }
```